render/render.go

```diff
@@ -354,3 +356,62 @@ func (f Filter) Render(rpt report.Report) RenderableNodes {
 	}
 	return output
 }
+
+// FilterSystem is a Renderer which filters out system nodes.
+func FilterSystem(r Renderer) Renderer {
+	return CustomRenderer{
+		RenderFunc: nonSystem,
+		Renderer:   r,
+	}
+}
+
+func nonSystem(input RenderableNodes) RenderableNodes {
+	// Deleted nodes also need to be cut as destinations in adjacency lists.
+	// So, we need to count all nodes to reject, before effecting mutation.
+	reject := map[string]struct{}{}
+	for id, node := range input {
+		containerName := node.Metadata[docker.ContainerName]
+		if _, ok := systemContainerNames[containerName]; ok {
+			reject[id] = struct{}{}
+		}
+		imageName := strings.SplitN(node.Metadata[docker.ImageName], ":", 2)[0] // :(
+		if _, ok := systemImageNames[imageName]; ok {
+			reject[id] = struct{}{}
+		}
+	}
+
+	output := RenderableNodes{}
+	for id, node := range input {
+		// Check for completely rejected node.
+		if _, ok := reject[id]; ok {
+			continue
+		}
+
+		// Accepted node, but maybe some bad adjacencies.
+		// Cut those before continuing.
+		newAdjacency := make(report.IDList, 0, len(node.Adjacency))
+		for _, dstID := range node.Adjacency {
+			if _, ok := reject[dstID]; ok {
+				continue
+			}
+			newAdjacency = newAdjacency.Add(dstID)
+		}
+
+		// Good.
+		node.Adjacency = newAdjacency
+		output[id] = node
+	}
+	return output
+}
+
+var systemContainerNames = map[string]struct{}{
+	"weavescope": struct{}{},
+	"weavedns":   struct{}{},
+	"weave":      struct{}{},
+}
+
+var systemImageNames = map[string]struct{}{
+	"weaveworks/scope":    struct{}{},
+	"weaveworks/weavedns": struct{}{},
+	"weaveworks/weave":    struct{}{},
+}
```

Comment:
hmmm... instead of hardcoding a bunch of images / container names, I thought we were going to use labels?

Also, you forgot weaveexec and weaveproxy.